pkg/agent/agent_readiness.go

```diff
@@ -53,6 +54,33 @@ func AllReachedGoalState(sts appsv1.StatefulSet, podGetter pod.Getter, desiredMe
 	return true, nil
 }
 
+func GetAllDesiredMembersPodState(sts appsv1.StatefulSet, podGetter pod.Getter, desiredMemberCount, targetConfigVersion int, log *zap.SugaredLogger) ([]PodState, error) {
```

Comment:
Please add a comment to this method as to what it does

Reply:
Added. 
Btw. I also spotted that we don't need statefulSet here - only namespace and name. So I have also refactored it a bit. But I haven't changed the signature of `AllReachedGoalState` as it is used in enterprise.